e2e/nomostest/new.go

```diff
@@ -203,6 +202,11 @@ func SharedTestEnv(t nomostesting.NTB, opts *ntopts.New) *NT {
 		}
 	})
 
+	// re-init the GitProvider.
+	// The setPortCallback must be registered for this NT so that it initializes
+	// the correct repositories.
+	nt.initGitProvider()
```

Comment:
Is it deliberate that this is set after the Cleanup is registered? Construction of the provider and forwarder doesn't look like it does anything that needs cleanup.

Reply:
In fact, it's before Cleanup in FreshTestEnv, so it should probably be before cleanup here too.

Reply:
`Cleanup` is essentially a no-op until the end of the test so I don't think it really matters, but I can change the ordering

Reply:
reordered